### PR TITLE
Initial chunk of manpages  for the buildah project

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Second chunk of manpages.  I think they're all up to current changes, but several need more/better examples.  I've also included a new common routine that will check to see if a string is a prefix in one or more strings within a slice.  My thinking is we can use that for the delete (or other commands) to do short name checking.  It's currently not as efficient as I'd like, but thought I'd get the general concept approved before trying to make the search faster.  It is not currently being called by any other function.

### CODE REVIEW

**Comment by rhatdan** [comment]
bot, add author to whitelist

File: docs/buildah-add.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+**buildah** **add** **containerID** **SRC** [**DEST**] 
+
+## DESCRIPTION
+Adds the contents of a file, URL, or a directory to a container's working directory.  If a local file appears to be an archive, its contents are extracted and added instead of the archive file itself.  If a destination is not specified, the location of the source will be used for the destination.  
+
+## EXAMPLE
+**buildah add containerID '/myapp/app.conf' '/myapp/app.conf' **
+**buildah add containerID '/home/myuser/myproject.go' **
+**buildah add containerID '/home/myuser/myfiles.tar' '/tmp' **
+**buildah add containerID '/tmp/workingdir' '/tmp/workingdir' **
```
Comment (rhatdan) [inline docs/buildah-add.md]: Need an example of multiple inputs going to a single output

buildah add containerID 'passwd' 'certs.d' /etc

File: docs/buildah-add.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+**buildah add containerID 'https://github.com/projectatomic/buildah' '/tmp' **
+
+## SEE ALSO
+buildah-commit(1)
```
Comment (rhatdan) [inline docs/buildah-add.md]: Lets just add a buildah(1) man page  and only refer to the others there.  Perhaps refer to related man pages, but not all, otherwise we will always be updating them all.
buildah-rmi and buildah-images, buildah-from, buildah-commit are related.

File: docs/buildah-from.md [hunk 3]
```diff
@@ -1,17 +1,26 @@
 ## buildah-from "March 2017"
 
 ## NAME
-buildah from - Creates a container image based on the supplied image name. If the supplied image name is "scratch" a new default empty container image will be created. 
+buildah from - Creates a new working container, either from scratch or using a specified image as a starting point. 
 
 
 ## SYNOPSIS
-**buildah** **from** --image *imagename* [*options* [...]] 
+**buildah** **from** *imagename* [*mountpoint*] [*options* [...]]
```
Comment (rhatdan) [inline docs/buildah-from.md]: No mount point support yet if ever.

File: docs/buildah-rmi.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+## buildah-rmi "March 2017"
+
+## NAME
+buildah rmi - Removes an image or images.
```
Comment (rhatdan) [inline docs/buildah-rmi.md]: +buildah rmi - Removes one or more images.. 

Need a comment about what happens if the images are being used.

File: docs/buildah-run.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+## NAME
+buildah run - Run a command inside of the container. 
+
+
+## SYNOPSIS
+**buildah** **run** **containerID** [*command options* [...]] 
+
+## DESCRIPTION
+Runs a specified command using the container's root filesystem as a root filesystem, using configuration settings inherited from the container's image or as specified using previous calls to the config command
```
Comment (rhatdan) [inline docs/buildah-run.md]: This actually launches a container for the running of the command, so that it is somewhat safer then running a random command.  THe command runs inside of the container.

File: docs/buildah-add.md (hunk 1 above)
Comment (TomSweeneyRedHat) [inline docs/buildah-add.md]: I've added this multi input example.

File: docs/buildah-add.md (hunk 2 above)
Comment (TomSweeneyRedHat) [inline docs/buildah-add.md]: Ok, will add a buildah man page.  I'm on the fence about listing any of the other pages other than the new main buildah man page in each of these command pages.  Will ruminate.

File: docs/buildah-from.md (hunk 3 above)
Comment (TomSweeneyRedHat) [inline docs/buildah-from.md]: I've removed the ,[*mountpoint*] from buildah-from.

File: docs/buildah-rmi.md (hunk 4 above)
Comment (TomSweeneyRedHat) [inline docs/buildah-rmi.md]: Changed the title for rmi and have added the bit about failure on multiple images.  WIll add the same to rm.

File: docs/buildah-run.md (hunk 5 above)
Comment (TomSweeneyRedHat) [inline docs/buildah-run.md]: So if I change:  "Runs a specified command using the container's..." to "Launches a container and runs the specified command in that container using the container's...".  Would that address this concern?  I'll make it so, holler if it needs adjusting.

Comment (rhatdan) [inline docs/buildah-run.md]: :+1:

File: cmd/buildah/common.go [hunk 6]
```diff
@@ -57,3 +59,27 @@ func openBuilder(store storage.Store, name string) (builder *buildah.Builder, er
 func openBuilders(store storage.Store) (builders []*buildah.Builder, err error) {
 	return buildah.OpenAllBuilders(store)
 }
+
+/****
+* Returns up to the first two names in the slice whose value starts with
+* the same value as the shortName.
+*****/
+func findShortName(names []string, shortName string) []string {
```
Comment (rhatdan) [inline cmd/buildah/common.go]: This looks like it belongs to a different patch set.  Accidently added?

**Comment by rhatdan** [comment]
You need to rebase your patches and merge them together, then do a force push.
git rebase -i origin
git push twsweeney --force

File: cmd/buildah/common.go (hunk 6 above)
Comment (TomSweeneyRedHat) [inline cmd/buildah/common.go]: No, I just tried to sleeze it through.  I was working on it the other night and decided to put the convenience function in play.  If you'd rather I do it separately I can pull it.  It's not being called/used yet.  I do need to bump the '> 0' up to '> 1' though, that was a cut/paste error while I was testing it.

Comment (rhatdan) [inline cmd/buildah/common.go]: Yes keep it out, and add it to the kpod rm and kpod rmi, patch set you are working on.

File: cmd/buildah/common.go [hunk 7]
```diff
@@ -3,6 +3,8 @@ package main
 import (
 	"fmt"
 	"os"
+        "sort"
+        "strings"
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/common.go]: Ugh.  Pulled the function, but forgot about the imports.

File: docs/buildah.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    * Create a working container, either from scratch or using an image as a starting point.
+    * Mount the working container's root filesystem for manipulation.
+    * Unmount the working container's root filesystem.
+    * Use the updated contents of the container's root filesystem as a filesystem layer to create a new image.
+    * Delete a working container or it's image.
+
+
+## SEE ALSO
+buildah-add(1)
```
Comment (rhatdan) [inline docs/buildah.md]: We need a one line comment on each one of these.

`buildah add`
          buildah add command adds the specified content to the container id.
...

**Comment by rhatdan** [comment]
You meed to update the Makefile to install convert and install these man pages.

**Comment by TomSweeneyRedHat** [comment]
Checking with Nalin about the test errors.

**Comment by TomSweeneyRedHat** [comment]
LGTM now after tweaking per Nalin's suggestions.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ b179ead

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot r+ b179ead

**Comment by nalind** [comment]
@rh-atomic-bot retry

**Comment by nalind** [comment]
@rh-atomic-bot r+ b179ead947c7a07e97ff3ac0ea2dc049c64cccd2

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by TomSweeneyRedHat** [comment]
@rhatdan FWIW, Nalin and I were both not able to get the homu test to re-run last week.

**Comment by nalind** [comment]
@rh-atomic-bot retry
